ocdata/core/adsorbate_slab_config.py

```diff
@@ -0,0 +1,492 @@
+        slab_elements = slab_c2.get_chemical_symbols()
+        projected_points = self._get_projected_points(adsorbate_c, slab_c2, unit_normal)
+
+        pairs = list(product(list(range(len(adsorbate_c))), list(range(len(slab_c2)))))
+
+        combos_to_check = []
+        for combo in pairs:
+            distance = np.linalg.norm(
+                projected_points["ads"][combo[0]] - projected_points["slab"][combo[1]]
+            )
+            radial_distance = (
+                covalent_radii[atomic_numbers[adsorbate_elements[combo[0]]]]
+                + covalent_radii[atomic_numbers[slab_elements[combo[1]]]]
+            )
+            if distance <= radial_distance:
+                combos_to_check.append(
+                    [combo, radial_distance, slab_c2.positions[combo[1]]]
+                )
+        return combos_to_check
+
+    def _get_projected_points(
+        self, adsorbate_c: ase.Atoms, slab_c2: ase.Atoms, unit_normal: np.ndarray
+    ):
+        """
+        Find the x and y coordinates of each atom projected onto the surface plane.
+        Args:
+            adsorbate_c (ase.Atoms): A copy of the adsorbate with coordinates at the site
+            slab_c2 (ase.Atoms): A copy of the slab with atoms wrapped s.t. things are centered
+                about the site
+            unit_normal (np.ndarray): the unit vector normal to the surface
+
+        Returns:
+            (dict): {"ads": [[x1, y1], [x2, y2], ...], "slab": [[x1, y1], [x2, y2], ...],}
+        """
+        projected_points = {"ads": [], "slab": []}
+        point_on_surface = slab_c2.cell[0]
+        for atom_position in adsorbate_c.positions:
+            v_ = atom_position - point_on_surface
+            projected_point = point_on_surface + (
+                v_
+                - (np.dot(v_, unit_normal) / np.linalg.norm(unit_normal) ** 2)
+                * unit_normal
+            )
+            projected_points["ads"].append(projected_point[0:2])
+
+        for atom_position in slab_c2.positions:
+            v_ = atom_position - point_on_surface
+            projected_point = point_on_surface + (
+                v_
+                - (np.dot(v_, unit_normal) / np.linalg.norm(unit_normal) ** 2)
+                * unit_normal
+            )
+            projected_points["slab"].append(projected_point[0:2])
+        return projected_points
+
+    def get_metadata_dict(self, ind):
+        """
+        Returns a dict containing the atoms object and metadata for
+        one specified config, used for writing to files.
+        """
+        return {
+            "adsorbed_slab_atomsobject": self.atoms_list[ind],
+            "adsorbed_slab_metadata": {
+                "bulk_id": self.slab.bulk.src_id,
+                "millers": self.slab.millers,
+                "shift": self.slab.shift,
+                "top": self.slab.top,
+                "smiles": self.adsorbate.smiles,
+            },
```

Comment:
I got rid of the sampling string and binding site (see https://github.com/Open-Catalyst-Project/Open-Catalyst-Dataset/blob/cf919723dbcf13bb5949701f4c3133e2f01b1914/ocdata/combined.py#L579 for previous version). I think they were unused but let me know if they should be kept.